roads.mss

```diff
@@ -2667,6 +2667,428 @@ come in as well.
+    [type='bridleway'][bicycle='yes'],
+    [type='bridleway'][bicycle='designated'] {
+      line-color: lighten(@path-fill,@surfaceLighterDarker1);
+    }
+    [type='path'][access='no'][bicycle=null],
+    [type='path'][bicycle='no'] {
+      line-color: lighten(@path-nobicycle,@surfaceLighterDarker1);
+    }
+    [type='service'],
+    [type='track'],
+    [type='tertiary'],
+    [type='living_street'],
+    [type='road'],
+    [type='unclassified'],
+    [type='residential'],
+    [type='tertiary_link'],
+    [type='secondary_link'],
+    [type='primary_link'],
+    [type='trunk_link'],
+    [type='motorway_link'],
+    [type='primary'],
+    [type='secondary'],
+    [type='motorway'],
+    [type='trunk'] {
+      line-cap: round;
+      line-join: round;
+    }
+    [type='cycleway'],
+    [type='pedestrian'],
+    [type='bridleway'],
+    [type='footway'],
+    [type='path'] {
+      line-join: round;
+    }
+
+    /* low maxspeed roads are bike friendly */
+    [type='pedestrian'],
+    [type='living_street'],
+    [type='service'],
+    [type='tertiary'],
+    [type='road'],
+    [type='unclassified'],
+    [type='residential'],
+    [type='tertiary_link'],
+    [type='secondary_link'],
+    [type='secondary'],
+    [type='primary_link'],
+    [type='primary'],
+    {
+      [maxspeed_kmh<33]
+      {
+        line-color: lighten(@speed32-fill,@surfaceLighterDarker1);
+      }
+      [maxspeed_kmh<21]
+      {
+        line-color: lighten(@speed20-fill,@surfaceLighterDarker1);
+      }
+      [maxspeed_kmh<10]
+      {
+        line-color: lighten(@speedWalk-fill,@surfaceLighterDarker1);
+      }
+    }
+
+    [zoom=16] {
+      [type='primary']     { line-width: @rdz16_primary; }
+      [type='secondary']     { line-width: @rdz16_secondary; }
+      [type='tertiary']    { line-width: @rdz16_tertiary; }
+      [type='living_street']    { line-width: @rdz16_living_street; }
+      [type='road']    { line-width: @rdz16_road; }
+      [type='unclassified']    { line-width: @rdz16_unclassified; }
+      [type='residential']    { line-width: @rdz16_residential; }
+      [type='tertiary_link']    { line-width: @rdz16_tertiary_link; }
+      [type='secondary_link']    { line-width: @rdz16_secondary_link; }
+      [type='primary_link']    { line-width: @rdz16_primary_link; }
+      [type='trunk_link']    { line-width: @rdz16_trunk_link; }
+      [type='motorway_link']    { line-width: @rdz16_motorway_link; }
+      [type='service']      { line-width: @rdz16_service; }
+      [type='track']      { line-width: @rdz16_track; }
+      [type='pedestrian']   { line-width: @rdz16_pedestrian; }
+      [type='bridleway']   { line-width: @rdz16_bridleway; }
+      [type='footway']   { line-width: @rdz16_footway; }
+      [type='path'],
+      [type='footway'][bicycle='yes'],
+      [type='footway'][bicycle='designated'],
+      [type='bridleway'][bicycle='yes'],
+      [type='bridleway'][bicycle='designated'] {
+        line-width: @rdz16_path;
```

Comment:
All the line widths here could probably be put in a single place with all the line widths below. This would avoid (a bit of) code duplication.

Reply:
But they must not be rendered if the surface quality good...